### ISSUE TITLE
Automation API: Support "pulumi new" operations

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi#7320]
## Affected feature
Automation API: 
Currently the automation api does not support "pulumi new" commands.
However to support a self-serve portal use-case, one needs to be able to create user-specific projects and stacks as they are requested.
Having functionality to be able to do the equivalent of "pulumi new" as part of the automation api would help address this use-case.

**Comment by leezen** [comment]
Stacks can be created via the automation API, so I'm wondering what functionality is missing that would be helpful here? Is it having a simpler way to stamp out `Pulumi.yaml` as part of creating a local project?

**Comment by MitchellGerdisch** [comment]
Yes, that's basically the use-case. To be able to create a project and then a related stack (or stacks) all from the automation api.

**Comment by orionstudt** [comment]
I am currently creating Projects & Stacks from the Automation API. When you create a stack you can also provide `ProjectSettings` which will define the root `Pulumi.yaml`. I'm curious what functionality you are missing

**Comment by RobbieMcKinstry** [comment]
This might be a good candidate for an experimental feature if we were to add a gRPC boundary to AutoAPI.
Since this is a net-new feature, we could test it out with the gPRC boundary without impacting existing AutoAPI methods.
